src/main/java/org/broadinstitute/hellbender/tools/gvs/common/WeightedSplitIntervals.java

```diff
@@ -0,0 +1,219 @@
+@CommandLineProgramProperties(
+        summary = "Split intervals into equally weight sub-interval files",
+        oneLineSummary = "Split intervals into equally weight sub-interval files",
+        programGroup = ShortVariantDiscoveryProgramGroup.class
+)
+
+public class WeightedSplitIntervals extends GATKTool {
+
+    public static final String SCATTER_COUNT_SHORT_NAME = "scatter";
+    public static final String SCATTER_COUNT_LONG_NAME = "scatter-count";
+
+    public static final String INTERVAL_FILE_PREFIX_FULL_NAME = "interval-file-prefix";
+    public static final String INTERVAL_FILE_EXTENSION_FULL_NAME = "extension";
+
+    public static final String DEFAULT_PREFIX = "";
+    public static final String PICARD_INTERVAL_FILE_EXTENSION = "interval_list";
+    public static final String DEFAULT_EXTENSION = "-scattered." + PICARD_INTERVAL_FILE_EXTENSION;
+
+    public static final String INTERVAL_NUMBER_OF_DIGITS_FULL_NAME = "interval-file-num-digits";
+    public static final int DEFAULT_NUMBER_OF_DIGITS = 4;  //to preserve backward compatibility
+
+    public static final String WEIGHTS_BED_FILE_FULL_NAME = "weight-bed-file";
+    public static final String DEFAULT_WEIGHT_FULL_NAME = "default-weight-per-base";
+
+    @Argument(fullName = SCATTER_COUNT_LONG_NAME, shortName = SCATTER_COUNT_SHORT_NAME,
+            doc = "scatter count: number of output interval files to split into", optional = true)
+    private int scatterCount = 1;
+
+    @Argument(doc = "The directory into which to write the scattered interval sub-directories.",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME)
+    public File outputDir;
+
+    @Argument(doc = "Prefix to use when writing interval files", fullName = INTERVAL_FILE_PREFIX_FULL_NAME, optional = true)
+    public String prefix = DEFAULT_PREFIX;
+
+    @Argument(doc = "Extension to use when writing interval files", fullName = INTERVAL_FILE_EXTENSION_FULL_NAME, optional = true)
+    public String extension = DEFAULT_EXTENSION;
+
+    @Argument(doc = "Number of digits to use when writing interval files", fullName = INTERVAL_NUMBER_OF_DIGITS_FULL_NAME, minValue = 1, optional = true)
+    public int numDigits = DEFAULT_NUMBER_OF_DIGITS;
+
+    @Argument(doc = "Scattered interval files do not contain intervals from multiple contigs.  This is applied after the initial scatter, so that the requested scatter count is a lower bound on the number of actual scattered files.", fullName = "dont-mix-contigs", optional = true)
+    public boolean dontMixContigs = false;
+
+    @Argument(doc = "BED file of genomic weights, represented by the score field", fullName = WEIGHTS_BED_FILE_FULL_NAME, optional = false)
+    public GATKPath weightsBedFile;
+
+    @Argument(doc = "Default weight (per base) to use if weight not found in BED file", fullName = DEFAULT_WEIGHT_FULL_NAME, optional = true)
+    public long defaultWeightPerBase = 0L;
```

Comment:
Why is this long when other default weights / base are measured in float?  It should probably be either float or int.  It's not broken, it's just weird.

Reply:
Also, I still think 0 default weight is asking for trouble.  You'll end up with everything in 1 giant interval.

Reply:
It should be a float, will change.

On the zero versus one… If you make it one then you're basically weighting be the length of the interval.  Whereas if you don't supply a weight for something we want it to say 'this has no cost' not 'make the cost proportional to the length'